tools/subgraph-example/contracts/GravatarRegistry.sol

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+contract GravatarRegistry {
+  constructor () {
+    setMythicalGravatar();
+  }
+
+  event NewGravatar(uint id, address owner, string displayName, string imageUrl);
+  event UpdatedGravatar(uint id, address owner, string displayName, string imageUrl);
+
+  struct Gravatar {
+    address owner;
+    string displayName;
+    string imageUrl;
+  }
+
+  Gravatar[] public gravatars;
+
+  mapping (uint => address) public gravatarToOwner;
+  mapping (address => uint) public ownerToGravatar;
+
+  function createGravatar(string calldata _displayName, string calldata _imageUrl) public {
+    require(ownerToGravatar[msg.sender] == 0);
+    gravatars.push(Gravatar(address(msg.sender), _displayName, _imageUrl));
+    uint id = gravatars.length - 1;
+
+    gravatarToOwner[id] = msg.sender;
+    ownerToGravatar[msg.sender] = id;
+
+    emit NewGravatar(id, msg.sender, _displayName, _imageUrl);
+  }
+
+  function getGravatar(address owner) public view returns (string memory, string memory) {
+    uint id = ownerToGravatar[owner];
+    return (gravatars[id].displayName, gravatars[id].imageUrl);
+  }
+
+  function updateGravatarName(string calldata _displayName) public {
+    require(ownerToGravatar[msg.sender] != 0);
+    require(msg.sender == gravatars[ownerToGravatar[msg.sender]].owner);
+
+    uint id = ownerToGravatar[msg.sender];
+
+    gravatars[id].displayName = _displayName;
+    emit UpdatedGravatar(id, msg.sender, _displayName, gravatars[id].imageUrl);
+  }
+
+  function updateGravatarImage(string calldata _imageUrl) public {
+    require(ownerToGravatar[msg.sender] != 0);
+    require(msg.sender == gravatars[ownerToGravatar[msg.sender]].owner);
+
+    uint id = ownerToGravatar[msg.sender];
+
+    gravatars[id].imageUrl =  _imageUrl;
+    emit UpdatedGravatar(id, msg.sender, gravatars[id].displayName, _imageUrl);
+  }
+
+  // the gravatar at position 0 of gravatars[]
+  // is fake
+  // it's a mythical gravatar
+  // that doesn't really exist
+  // dani will invoke this function once when this contract is deployed
```

Comment:
`anyone` or `someone` may invoke this post deployment

Reply:
I updated the contract, in the original contracts the function looks like this 
```
  // the gravatar at position 0 of gravatars[]
  // is fake
  // it's a mythical gravatar
  // that doesn't really exist
  // dani will invoke this function once when this contract is deployed
  // but then no more
  function setMythicalGravatar() public {
    require(msg.sender == 0x8d3e809Fbd258083a5Ba004a527159Da535c8abA);
    gravatars.push(Gravatar(0x0, " ", " "));
  }
```

I updated it to be executed once in the constructor and no-one can execute it outside of the contract